tools/devops/automation/templates/common/mac-agent-logs.yml

```
@@ -49,14 +53,26 @@ steps:
 
 # only upload when told so
 - ${{ if eq(parameters.uploadLogs, true) }}:
-  - task: PublishPipelineArtifact@1
-    displayName: 'Publish Artifact: Agent logs'
-    condition: always()
-    inputs:
-      # add or not the working dir to the upload
-      ${{ if eq(parameters.workingDirectory, '') }}:
-        targetPath: ${{ parameters.outputPath }}
-      ${{ else }}:
-        targetPath: ${{ parameters.workingDirectory }}/${{ parameters.outputPath }} 
-      artifactName: '${{ parameters.uploadPrefix }}$(Agent.Name)-$(System.JobName)-$(System.JobId)-${{ parameters.outputPath }}'
-    continueOnError: true
+  - ${{ if eq(parameters.use1ES, true) }}:
+    - task: 1ES.PublishPipelineArtifact@1
+      displayName: 'Publish Artifact: Agent logs'
+      condition: always()
+      inputs:
+        ${{ if eq(parameters.workingDirectory, '') }}:
+          path: ${{ parameters.outputPath }}
+        ${{ else }}:
+          path: ${{ parameters.workingDirectory }}/${{ parameters.outputPath }} 
+        artifact: '${{ parameters.uploadPrefix }}$(Agent.Name)-$(System.JobName)-$(System.JobId)-${{ parameters.outputPath }}'
+      continueOnError: true
+  - ${{ else }}:
+    - task: PublishPipelineArtifact@1
+      displayName: 'Publish Artifact: Agent logs'
+      condition: always()
+      inputs:
+        # add or not the working dir to the upload
+        ${{ if eq(parameters.workingDirectory, '') }}:
+          targetPath: ${{ parameters.outputPath }}
+        ${{ else }}:
+          targetPath: ${{ parameters.workingDirectory }}/${{ parameters.outputPath }} 
+        artifactName: '${{ parameters.uploadPrefix }}$(Agent.Name)-$(System.JobName)-$(System.JobId)-${{ parameters.outputPath }}'
+      continueOnError: true
```

It doesn't work to simplify the condition to have only the single different line inside the if?

```suggestion
  - ${{ if eq(parameters.use1ES, true) }}:
    - task: 1ES.PublishPipelineArtifact@1
  - ${{ else }}:
    - task: PublishPipelineArtifact@1
    displayName: 'Publish Artifact: Agent logs'
    condition: always()
    inputs:
      ${{ if eq(parameters.workingDirectory, '') }}:
        path: ${{ parameters.outputPath }}
      ${{ else }}:
        path: ${{ parameters.workingDirectory }}/${{ parameters.outputPath }} 
      artifact: '${{ parameters.uploadPrefix }}$(Agent.Name)-$(System.JobName)-$(System.JobId)-${{ parameters.outputPath }}'
    continueOnError: true
```

---

I wish, that would work if the 1ES people chose to make the task parameters compatible with the default task. But they decided to do:

* targetPath -> path

Artefact does work AFAIK: https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/reference/publish-pipeline-artifact-v1?view=azure-pipelines but we would make it worse due to targetPath.